lib/datasource/npm.js

```diff
@@ -206,6 +199,10 @@ async function getDependencyInner(name, retries = 5) {
       versions: {},
       'renovate-config': latestVersion['renovate-config'],
     };
+    if (latestVersion.deprecated) {
```

Comment:
Is it possible that the version pointed to by `latest` dist-tag would be deprecated, but an earlier, perhaps major version, would still receive bug fixes (maybe it's supported by the developer)?

Reply:
> Uses logic that if latest version is deprecated then the package is deprecated.

Nevermind. You answer this question above in your pull request comment.

Reply:
This seems to be the logic that npm uses. FYI, I have not found an example where there is a `deprecated` field at the "top-level" as you wrote in requirements, so I have inferred the logic.

Reply:
Also, I added a commit yesterday to just add a log message whenever such a case was found and looked through them manually. I didn't check *every single one*, but checked dozens and they were all correct.